elide-core/src/main/java/com/yahoo/elide/security/FilterExpressionCheck.java

```diff
@@ -0,0 +1,161 @@
+    public final boolean ok(T object, RequestScope requestScope, Optional<ChangeSpec> changeSpec) {
+        FilterExpression filterExpression = getFilterExpression(requestScope);
+        return applyFilterExpressionToObject(object, filterExpression);
+    }
+
+    private boolean applyFilterExpressionToObject(T object, FilterExpression filterExpression) {
+        if (filterExpression == null) {
+            return true;
+        }
+        if (AndFilterExpression.class.isAssignableFrom(filterExpression.getClass())) {
+            AndFilterExpression andFilterExpression = (AndFilterExpression) filterExpression;
+            FilterExpression left = andFilterExpression.getLeft();
+            FilterExpression right = andFilterExpression.getRight();
+            return applyFilterExpressionToObject(object, left) && applyFilterExpressionToObject(object, right);
+        }
+        if (OrFilterExpression.class.isAssignableFrom(filterExpression.getClass())) {
+            OrFilterExpression orFilterExpression = (OrFilterExpression) filterExpression;
+            FilterExpression left = orFilterExpression.getLeft();
+            FilterExpression right = orFilterExpression.getRight();
+            return applyFilterExpressionToObject(object, left) || applyFilterExpressionToObject(object, right);
+        }
+        if (NotFilterExpression.class.isAssignableFrom(filterExpression.getClass())) {
+            NotFilterExpression notFilterExpression = (NotFilterExpression) filterExpression;
+            FilterExpression negation = notFilterExpression.getNegated();
+            return !applyFilterExpressionToObject(object, negation);
+        }
+        if (Predicate.class.isAssignableFrom(filterExpression.getClass())) {
+            Predicate predicate = (Predicate) filterExpression;
+            return applyPredicateToObject(object, predicate);
+        }
+        return false;
+    }
+
+    private boolean applyPredicateToObject(T object, Predicate predicate) {
+        Predicate.PathElement path = predicate.getPath().get(predicate.getPath().size() - 1);
+        Class type = path.getType();
+        //If predicate is not related to object, vacuously true.
+        if (!type.isAssignableFrom(object.getClass())) {
+            return true;
+        }
+        Class fieldType = path.getFieldType();
+        String fieldName = path.getFieldName();
+        //same reason as above.
+        if (fieldName == null) {
+            return false;
+        }
+        try {
+            Object fieldValue = getFieldByName(object, type, fieldName);
+            Object expectedValue = predicate.getValues().get(predicate.getValues().size() - 1);
+            Operator op = predicate.getOperator();
+            return handleByOperatorCases(op, fieldValue, expectedValue);
+        } catch (NoSuchFieldException e) {
+            return true;
+        }
+
+    }
+
+    private boolean handleByOperatorCases(Operator op, Object fieldValue, Object predicateConstrain) {
```

Comment:
This should already be implemented inside the Operator.  You shouldn't have to reimplement the ability to evaluate predicates.